lib/graphql/schema/union.rb

```diff
@@ -5,13 +5,13 @@ class Union < GraphQL::Schema::Member
       extend GraphQL::Schema::Member::AcceptsDefinition
 
       class << self
-        def possible_types(*types)
+        def possible_types(*types, ctx: nil)
           if types.any?
             @possible_types = types
           else
             all_possible_types = @possible_types || []
             all_possible_types += super if defined?(super)
-            all_possible_types.uniq
+            ctx ? filter_possible_types(all_possible_types.uniq, ctx) : all_possible_types.uniq
```

Comment:
🤔 it might be better to borrow an existing pattern in the gem to create a `GraphQL::Query::NullContext` when theres no ctx and make the old method signature deprecated.

Especially here `ctx` isn't even used; so `filter_possible_types` can be called in either case with the same output.

Reply:
This would apply in a few different places I guess.

Reply:
Yah I need a better spot to check for nil context, I didn't want to leave it up to the user. `NullContext` looks like it is what I want though, thanks!